### DISCUSSION TITLE
Klipper and Moonraker are not compatible anymore.

### DISCUSSION THREAD

**Discussion by Makej4f** [Arksine/moonraker#527] [Q&A]
````
Moonraker-Warnungen gefunden.
[server]: Option 'enable_debug_logging' is deprecated, see the configuration documention at https://moonraker.readthedocs.io/en/latest/configuration/
[database]: Option 'database_path' is deprecated, see the configuration documention at https://moonraker.readthedocs.io/en/latest/configuration/
[file_manager]: Option 'config_path' is deprecated, see the configuration documention at https://moonraker.readthedocs.io/en/latest/configuration/
[file_manager]: Option 'log_path' is deprecated, see the configuration documention at https://moonraker.readthedocs.io/en/latest/configuration/
Failed to link subfolder 'database' to source path '/home/user/moonraker_database'. The requested path is not a valid directory.
```

The data path (klipper_config and printer_config )is not matching together after the last update from 14.10.2022.

Do you have any idea how to fix this? I executed already `data-path-fix.sh` from moonraker.

**Comment by Arksine** [comment]
From reading this snippet it appears you have something like the following in `moonraker.conf`:
```
[database]
database_path: /home/user/moonraker_database
```

However no such database exists on your system.  Is this a new installation?  Can you download `moonraker.log` and attach it?

**Comment by Dalpek** [comment]
I have 2 instances of moonraker and I'm having the same issue. How can I edit data-path-fix,sh to work for 2 instances?

**Reply by Arksine** [reply]
Same questions...are these new installations?  Can you download `moonraker.log` from one of the instances and attach it?

**Reply by Dalpek** [reply]
Hello,
Thanks for the help!
Yesterday I upgraded and Mainsail broke. I decided to reinstall everything from zero. Including the pi operating system. It's a new installation. After the re-installation, I got some error messages, which I corrected by editing some of the config files. Now, Mainsail has no errors, but all the paths are screwed up. I can not print a file, see the current log, or the config files via mainsail. I have to do it via SSH.
[moonraker-Ender3S1pro.log_20221017150241.log](https://github.com/Arksine/moonraker/files/9804156/moonraker-Ender3S1pro.log_20221017150241.log)
![Screenshot 2022-10-17 153150](https://user-images.githubusercontent.com/76981707/196277459-ffafdd30-372b-4ebb-914b-fed3e2f270a4.jpg)
![Screenshot 2022-10-17 153246](https://user-images.githubusercontent.com/76981707/196277581-4c0008db-01db-4570-9f69-f49ef1b02945.jpg)

**Reply by Arksine** [reply]
Assuming your gcode files are located at `~/gcode_files` I would suggest the following via ssh:
```
sudo service moonraker-Ender3S1pro stop
cd ~/moonraker-Ender3S1pro_data
rm -rf gcodes
ln -s ~/gcode_files gcodes
rm -rf config
ln -s ~/klipper_config/Ender3S1pro config
rm -rf logs
ln -s ~/klipper_logs logs
sudo service moonraker-Ender3S1pro start
```

Do the same with the other instance, but replace `Ender3S1pro` with the appropriate instance name.

**Reply by Dalpek** [reply]
WOW!!
I hadn't though of making symbolic links, thanks! It worked perfectly on both instances!!

**Reply by Arksine** [reply]
Yeah, that is what `data-path-fix.sh` does, however it also reinstalls Moonraker which I didn't believe was necessary after looking at your log.

Moonraker's install validator also creates symbolic links, however it may have gotten confused by the unique instance names.  Automating the conversion of multiple instances is bit tricky given the numerous permutations available.

I'm glad its working.  Th3xxitus is working on an update to KIAUH, so in the future new installs should go smoothly.

**Reply by arielbernal** [reply]
data-path-fix.sh is outdated and broken

**Reply by Arksine** [reply]
I'm afraid you'll need to provide more information as to exactly how it is broken for me to provide any meaningful feedback.  To my knowledge it should still work for its designed purpose, ie: to convert older versions of MainsailOS and FluiddPi to the new data folder structure.  That said, it's purpose should be near EOL at this point. I would image most users have either performed the fix, installed a fresh version of Moonraker, or installed a newer version of the the OS.
